rclpy/CMakeLists.txt

```diff
@@ -80,18 +63,15 @@ add_library(rclpy_common SHARED
   src/rclpy_common/src/common.c
   src/rclpy_common/src/handle.c
 )
-target_link_libraries(rclpy_common
-  ${PythonExtra_LIBRARIES}
+target_link_libraries(rclpy_common PUBLIC
+  pybind11::pybind11
+  ${PYTHON_LIBRARIES}
+  rcl::rcl
+  rmw::rmw
 )
-target_include_directories(rclpy_common
-  PUBLIC
+target_include_directories(rclpy_common PUBLIC
   src/rclpy_common/include
 )
-ament_target_dependencies(rclpy_common
-  "rcl"
-  "rmw"
-  "PythonExtra"
-)
```

Comment:
Can you elaborate on why you're moving away from the `ament_target_dependencies` macro to directly using `target_link_libraries`? (same question for subsequent targets)

Reply:
`ament_target_dependencies()` is convenient when packages use the old-style standard CMake variables, but when there are exported CMake targets `ament_target_dependencies()` is no more convenient than using `target_link_libraries()` directly. I opened a ticket about it https://github.com/ament/ament_cmake/issues/292, and the switch here is me chipping away at it.

Reply:
Alright, thanks for the context.